db/migrate/20170601161315_add_anonymous_user_to_posts_without_userid.rb

```diff
@@ -0,0 +1,20 @@
+
+class AddAnonymousUserToPostsWithoutUserid < ActiveRecord::Migration[5.1]
+  def up
+    password = (0...15).map { ('a'..'z').to_a[rand(25)] }.join
+    user = User.create(name: "Anonymous", email: "anon@acebook.com", password: password)
+    Post.all.each { |post|
+      post.user_id ||= user.id
+      post.save
+    }
```

Comment:
I'd change the `{` to `do/end`s as it's a multiline block.

I'd also make `create` -> `create!` and `save` -> `save!` so they throw errors if anything doesn't work. Consider — if `create` failed, would you want it to throw an error and stop the migration? Or continue on regardless?